p2p/ipld/read_test.go

```diff
@@ -486,3 +484,45 @@ func generateRandomMessages(count, msgSize int) types.Messages {
 	}
 	return types.Messages{MessagesList: msgs}
 }
+
+func mockedIpfsAPI(t *testing.T) iface.CoreAPI {
+	ipfsNode, err := coremock.NewMockNode()
+	if err != nil {
+		t.Error(err)
+	}
+
+	// issue a new API object
+	ipfsAPI, err := coreapi.NewCoreAPI(ipfsNode)
+	if err != nil {
+		t.Error(err)
+	}
+
+	return ipfsAPI
+}
+
+func dataToIpfs(ctx context.Context, t *testing.T, api iface.CoreAPI, data [][]byte) cid.Cid {
+	batch := format.NewBatch(ctx, api.Dag().Pinning())
+
+	// create a random tree
+	tree, err := createNmtTree(ctx, batch, data)
```

Comment:
`createNmtTree` was made before `ErasuredNamespacedMerkleTree` was introduced. Ideally, we should use that instead, but I'm fine with leaving this as it was inherited from #232

Reply:
I rolled back `dataToIpfs`.